src/sentry/integrations/request_buffer.py

```diff
@@ -90,47 +100,14 @@ def add(self, count: str):
         if count not in VALID_KEYS:
             raise Exception("Requires a valid key param.")
 
-        other_count1, other_count2 = list(set(VALID_KEYS).difference([count]))[0:2]
         now = datetime.now().strftime("%Y-%m-%d")
 
-        buffer_key = self.integrationkey
+        buffer_key = self.integrationkey + now
         pipe = self.client.pipeline()
-
-        while True:
-            try:
-                pipe.watch(buffer_key)
-                recent_item_array = pipe.lrange(buffer_key, 0, 1)  # get first element from array
-                pipe.multi()
-                if len(recent_item_array):
-                    recent_item = self._convert_obj_to_dict(recent_item_array[0])
-                    if recent_item.get("date") == now:
-                        recent_item[f"{count}_count"] += 1
-                        pipe.lset(buffer_key, 0, json.dumps(recent_item))
-                    else:
-                        data = {
-                            "date": now,
-                            f"{count}_count": 1,
-                            f"{other_count1}_count": 0,
-                            f"{other_count2}_count": 0,
-                        }
-                        pipe.lpush(buffer_key, json.dumps(data))
-
-                else:
-                    data = {
-                        "date": now,
-                        f"{count}_count": 1,
-                        f"{other_count1}_count": 0,
-                        f"{other_count2}_count": 0,
-                    }
-                    pipe.lpush(buffer_key, json.dumps(data))
-                pipe.ltrim(buffer_key, 0, BUFFER_SIZE - 1)
-                pipe.expire(buffer_key, KEY_EXPIRY)
-                pipe.execute()
-                break
-            except WatchError:
-                continue
-            finally:
-                pipe.reset()
+        pipe.hincrby(buffer_key, count + "_count", 1)
+        pipe.expire(buffer_key, KEY_EXPIRY)
```

Comment:
nit: you can maybe include an `NX` optional param for the expire command https://redis.io/commands/expire/ and only set the expire once. In practical terms, it probably won't matter since you're only refreshing the ttl for this day and then when the its the next day, the previous day hash won't ever be modified again.